src/amplitude.js

```diff
@@ -137,146 +112,238 @@ Amplitude.prototype.init = function(apiKey, opt_userId, opt_config, callback) {
     }
   } catch (e) {
     utils.log(e);
+  } finally {
+    if (type(opt_callback) === 'function') {
+      opt_callback();
+    }
   }
+};
 
-  if (callback && type(callback) === 'function') {
-    callback();
+/**
+ * Parse and validate user specified config values and overwrite existing option value
+ * DEFAULT_OPTIONS provides list of all config keys that are modifiable, as well as expected types for values
+ * @private
+ */
+var _parseConfig = function _parseConfig(options, config) {
+  if (type(config) !== 'object') {
+    return;
   }
+
+  // verifies config value is defined, is the correct type, and some additional value verification
+  var parseValidateLoad = function parseValidateLoad(key, expectedType) {
+    var input_value = config[key];
+    if(input_value === undefined || !utils.validateInput(input_value, key + ' option', expectedType)) {
+      return;
+    }
+    if (expectedType === 'boolean') {
+      options[key] = !!input_value;
+    } else {
+      options[key] = (expectedType === 'string' && !utils.isEmptyString(input_value) && input_value) ||
+                     (expectedType === 'number' && config[key] > 0 && input_value) ||
+                     options[key];
```

Comment:
In the case of SDK options yes, they need to be positive. The numerical options are all thresholds and limits such as max event count, session timeout, upload batch size, event upload threshold, event upload period in millis. These would likely break the SDK with values of 0 or negative.